tests/feat-overwrite-init-containers/src/main/resources/application.properties

```diff
@@ -0,0 +1,2 @@
+dekorate.kubernetes.init-containers[0].name=foo
```

Comment:
Would it be possible to switch to a more realistic scenario, like having the user provide their own resources containing an  init  container and using property configuration to change things like env, image etc?

Reply:
This is a very fair request and indeed the first attempt was to replace the init container image configuration. The problem that I encountered is that the order of the init container was not respected. For example:

```
dekorate.kubernetes.init-containers[0].name=foo
dekorate.kubernetes.init-containers[0].command=sh
dekorate.kubernetes.init-containers[1].name=foo
dekorate.kubernetes.init-containers[1].command=sh updated
```

In this example, the `init-containers[1]` was applied first and then `init-containers[0]` which I think it's wrong. wdyt?

Reply:
I wasmostly asking to use an existing resource vs defining both init containers via props. Anyway,  this is no biggie. So let's move on.